teacher.rb

```diff
@@ -0,0 +1,14 @@
+require './person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(name = 'Unknown', age = nil, parent_permission: true)
+    super(name, age, parent_permission)
+    @specialization = specialization
+  end
```

Comment:
- [ ] The `specialization` should be passed as a parameter also.

- [ ] The `age` should not be optional

- [ ] The arrangement of arguments on line 7 should match the Person class.

- [ ] `parent_permission` should be well assigned in the super arguments.

  ```
  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @specialization= specialization
  end
  ```